lake/utils/test_infra.py

```diff
@@ -0,0 +1,210 @@
+                     stencil_valid=False):
+
+    lt_dut = LakeTop(interconnect_input_ports=in_ports,
+                     interconnect_output_ports=out_ports,
+                     stencil_valid=stencil_valid)
+
+    configs = lt_dut.get_static_bitstream(config_path, in_file_name, out_file_name)
+    configs_list = set_configs_sv(lt_dut, "configs.sv", get_configs_dict(configs))
+
+    magma_dut = kts.util.to_magma(lt_dut,
+                                  flatten_array=True,
+                                  check_multiple_driver=False,
+                                  optimize_if=False,
+                                  check_flip_flop_always_ff=False)
+
+    tester = fault.Tester(magma_dut, magma_dut.clk)
+
+    return lt_dut, configs, configs_list, magma_dut, tester
+
+
+def gen_test_lake(config_path,
+                  stream_path,
+                  in_file_name="input",
+                  out_file_name="output",
+                  in_ports=2,
+                  out_ports=2):
+
+    lt_dut, configs, configs_list, magma_dut, tester = \
+        base_lake_tester(config_path,
+                         in_file_name,
+                         out_file_name,
+                         in_ports,
+                         out_ports)
+
+    tester.circuit.clk = 0
+    tester.circuit.rst_n = 0
+    tester.step(2)
+    tester.circuit.rst_n = 1
+    tester.step(2)
+    tester.circuit.clk_en = 1
+    tester.eval()
+
+    # args are input ports, output ports
+    in_data, out_data, valids = generate_data_lists(stream_path, in_ports, out_ports)
+
+    for (f1, f2) in configs:
+        setattr(tester.circuit, f1, f2)
+
+    for i in range(len(out_data[0])):
+        for j in range(len(in_data)):
+            if i < len(in_data[j]):
+                setattr(tester.circuit, f"data_in_{j}", in_data[j][i])
+        # tester.circuit.data_in_0 = i
+
+        tester.eval()
+
+        for j in range(len(out_data)):
+            if i < len(out_data[j]):
+                if len(valids) != 0 and valids[i] == 1:
+                    getattr(tester.circuit, f"data_out_{j}").expect(out_data[j][i])
+                if len(valids) == 0:
+                    getattr(tester.circuit, f"data_out_{j}").expect(out_data[j][i])
+
+        tester.step(2)
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        tester.compile_and_run(target="verilator",
+                               directory=tempdir,
+                               flags=["-Wno-fatal"])
+
+
+def conv_3_3_args():
+    lc, ls = check_env()
+    config_path = lc + "conv_3_3_recipe/buf_inst_input_10_to_buf_inst_output_3_ubuf"
+    stream_path = ls + "conv_3_3_recipe/buf_inst_input_10_to_buf_inst_output_3_ubuf_0_top_SMT.csv"
+    in_file_name = "input_"
+    out_file_name = "output_2_"
+    return config_path, stream_path, in_file_name, out_file_name
+
+
+# not all of these tests/folders still exist,
+# update in_file_name and out_file_name args if needed
+def test_gaussian():
```

Comment:
can we just have the actual tests in tests/test_lake.py? The testing infrastructure can be here.

Reply:
these tests are all out of date and the paths don't exist for all of them,...would you be okay if I just delete them?

There is one test function in tests/test_lake.py now and that just takes the args (csv paths) for pytest and tests the actual tests there (right now, just conv_3_3).

Reply:
I deleted old tests to avoid confusion, lmk if you want that reverted.